pkgs/tools/misc/esphome/default.nix

```diff
@@ -36,11 +46,20 @@ python.pkgs.buildPythonApplication rec {
     pythonRelaxDepsHook
   ];
 
-  pythonRelaxDeps = true;
+  pythonRelaxDeps = [
+    "aioesphomeapi"
+    "cryptography"
+    "kconfiglib"
+    "tornado"
+    "zeroconf"
+  ];
```

Comment:
Why move from unpinning everything to only unpinning specific things?

Reply:
[Since the source code explicitly enforces that the pinned version matches exactly at runtime](https://github.com/esphome/esphome/blob/47a1710b1ed9da2304d0f17454f0b578d8d1554c/esphome/components/font/__init__.py#L115), either the dependency must be pinned to match the requirement, or the linked line in the source must be patched to relax its check alongside `pythonRelaxDeps` also relaxing the build-time pip requirement.

Reply:
Relax the linked line with `substituteInPlace --replace-fail`.

Reply:
Works for me, done.